pkg/router/controller/factory/factory.go

```diff
@@ -49,26 +82,133 @@ func (factory *RouterControllerFactory) Create(plugin router.Plugin) *controller
 	}
 }
 
+// CreateNotifier begins listing and watching against the API server for the desired route and endpoint
+// resources. It spawns child goroutines that cannot be terminated. It is a more efficient store of a
+// route system.
+func (factory *RouterControllerFactory) CreateNotifier(changed func()) RoutesByHost {
+	keyFn := cache.MetaNamespaceKeyFunc
+	routeStore := cache.NewIndexer(keyFn, cache.Indexers{"host": hostIndexFunc})
+	routeEventQueue := oscache.NewEventQueueForStore(keyFn, routeStore)
+	cache.NewReflector(&routeLW{
+		client:    factory.OSClient,
+		namespace: factory.Namespace,
+		field:     factory.Fields,
+		label:     factory.Labels,
+	}, &routeapi.Route{}, routeEventQueue, factory.ResyncInterval).Run()
+
+	endpointStore := cache.NewStore(keyFn)
+	endpointsEventQueue := oscache.NewEventQueueForStore(keyFn, endpointStore)
+	cache.NewReflector(&endpointsLW{
+		client:    factory.KClient,
+		namespace: factory.Namespace,
+		// we do not scope endpoints by labels or fields because the route labels != endpoints labels
+	}, &kapi.Endpoints{}, endpointsEventQueue, factory.ResyncInterval).Run()
+
+	go util.Until(func() {
+		for {
+			if _, _, err := routeEventQueue.Pop(); err != nil {
+				return
+			}
+			changed()
+		}
+	}, time.Second, util.NeverStop)
+	go util.Until(func() {
+		for {
+			if _, _, err := endpointsEventQueue.Pop(); err != nil {
+				return
+			}
+			changed()
```

Comment:
Would be good to add a parameter to the callback ```changed``` to indicate what changed - endpoint/route. 
This doesn't seem to be used currently, right?